### PR TITLE
convert parametersTypeToChars() to single parameter

### PR DESCRIPTION
**Pull request by WalterBright**
Nicely illustrates the improvement `ParameterList` begats.

### CODE REVIEW

File: src/dmd/hdrgen.h [hunk 1]
```diff
@@ -19,7 +19,7 @@ class Module;
 void genhdrfile(Module *m);
 void moduleToBuffer(OutBuffer *buf, Module *m);
 
-const char *parametersTypeToChars(Parameters *parameters, int varargs);
+const char *parametersTypeToChars(ParameterList pl);
```
Comment (thewilsonator) [inline src/dmd/hdrgen.h]: May need to have an overload depending on ABI issues, but otherwise looks good.

**Review by thewilsonator — APPROVED** [review]

File: src/dmd/hdrgen.h (hunk 1 above)
Comment (ibuclaw) [reply src/dmd/hdrgen.h]: Or even better, make it a ref parameter

File: src/dmd/hdrgen.d [hunk 2]
```diff
@@ -3189,25 +3189,29 @@ public:
         }
     }
 
-    void parametersToBuffer(Parameters* parameters, int varargs)
+    void parametersToBuffer(ParameterList pl)
```
Comment (ibuclaw) [inline src/dmd/hdrgen.d]: Looks like this needs updating as well to a C++ friendly signature, or made extern(D).

Comment (WalterBright) [reply src/dmd/hdrgen.d]: Does C++ code ever call it? And it's a POD struct.

**Pull request merged**